### DISCUSSION TITLE
Geo-location: track visitor countries [DONE - PRO FEATURE]

### DISCUSSION THREAD

**Discussion by dannyvankooten** [ibericode/koko-analytics#225] [Ideas]
For new visitors, increment the count for their respective country. Because this would need on-site geolocation (since we don't want to send IP addresses across the wire) this would most likely be best suited as a Pro feature.

**Comment by fancsali** [comment]
If you mean MaxMind geoip by 'on-site geolocation' I can't see the complexity. I am confident, there are PHP bindings for that - hence I can't see why that'd mean 'Pro feature'.

On the other hand, if you'd be prepared to maintain KokoAnalytcs' own GeoIP DB, that's another question...

**Reply by baden03** [reply]
Why limit to country? State was the missing feature that dismissed this plugin as a valid solution for us. Cities exist.

**Comment by dannyvankooten** [comment]
> If you mean MaxMind geoip by 'on-site geolocation' I can't see the complexity. I am confident, there are PHP bindings for that - hence I can't see why that'd mean 'Pro feature'.

The issue is mainly that MaxMind requires an account / license key to use their database and the database can't be included in the source code of the plugin because of its licensing. So we would need to add a mechanism for users to authenticate with MaxMind and then periodically refresh the database. It's of course possible, but a lot more complex than simply shipping the database file inside the plugin source. I'm not sure I am ready to support that on a free basis.

**Comment by fancsali** [comment]
Hm.

I was under the impression, that the legacy database is free – at least in
Debian you can (at least could) just say `apt-get install
geoip-whatever-legacy`, and those guys are pretty serious about licensing.

Perhaps the free version could come without database, and the user could
provide their own (by WP upload or OS path); the pro version would however
take care of the whole thing.

**Comment by dannyvankooten** [comment]
Geo-location (on a country level) is now included in [Koko Analytics Pro](https://www.kokoanalytics.com/pricing/).

<img width="1400" height="1080" alt="image" src="https://github.com/user-attachments/assets/54cd3986-66db-4d0a-bdf4-8422ea0bb3f8" />
